packages/react/src/components/ContentSwitcher/ContentSwitcher.js

```diff
@@ -141,25 +141,44 @@ export default class ContentSwitcher extends React.Component {
       ...other
     } = this.props;
 
+    const isIconOnly = React.Children.map(children, (child) => {
+      return child.type.displayName === 'IconSwitch';
+    }).every((val) => val === true);
+
     const classes = classNames(`${prefix}--content-switcher`, className, {
       [`${prefix}--content-switcher--light`]: light,
       [`${prefix}--content-switcher--${size}`]: size,
+      [`${prefix}--content-switcher--icon-only`]: isIconOnly,
     });
 
     return (
       <div {...other} className={classes} role="tablist">
-        {React.Children.map(children, (child, index) =>
-          React.cloneElement(child, {
-            index,
-            onClick: composeEventHandlers([
-              this.handleChildChange,
-              child.props.onClick,
-            ]),
-            onKeyDown: this.handleChildChange,
-            selected: index === this.state.selectedIndex,
-            ref: this.handleItemRef(index),
-          })
-        )}
+        {React.Children.map(children, (child, index) => {
+          if (child.type.displayName === 'IconSwitch') {
+            return React.cloneElement(child, {
+              index,
+              onClick: composeEventHandlers([
+                this.handleChildChange,
+                child.props.onClick,
+              ]),
+              onKeyDown: this.handleChildChange,
+              selected: index === this.state.selectedIndex,
+              ref: this.handleItemRef(index),
+              size,
+            });
+          } else {
+            return React.cloneElement(child, {
+              index,
+              onClick: composeEventHandlers([
+                this.handleChildChange,
+                child.props.onClick,
+              ]),
+              onKeyDown: this.handleChildChange,
+              selected: index === this.state.selectedIndex,
+              ref: this.handleItemRef(index),
+            });
+          }
+        })}
```

Comment:
@tay1orjones Do you think this is necessary? The only reason I've done this is so we don't pass `size` down to the standard `Switch` component, but I do not see any issues by passing it down, and it makes it harder to read. It ensures no extra props are sent down, but I think `size` is the only prop that needs to be sent down. The rest can be passed directly to `IconSwitch`.

Reply:
@tw15egan If `size` is passed down to `Switch`, it'll get lumped into `...other` which is spread onto a `button`. Would that generate one of those "invalid attribute on dom element" warnings?

Reply:
I didn't see any, but that may be because it isn't camel-cased?